src/server/identity/cmds/cmds.go

```diff
@@ -25,21 +29,32 @@ type connectorConfig struct {
 }
 
 func newConnectorConfig(conn *identity.IDPConnector) (*connectorConfig, error) {
-	config := connectorConfig{
+	srcConfig, err := identityutil.PickConfig(conn.Config, conn.JsonConfig)
+	if err != nil {
+		return nil, errors.EnsureStack(err)
+	}
+	config := map[string]interface{}{}
+	err = json.Unmarshal(srcConfig, &config)
+	if err != nil {
+		return nil, errors.EnsureStack(err)
+	}
+	connConfig := connectorConfig{
 		ID:      conn.Id,
 		Name:    conn.Name,
 		Type:    conn.Type,
 		Version: conn.ConfigVersion,
+		Config:  config,
 	}
-
-	if err := json.Unmarshal([]byte(conn.JsonConfig), &config.Config); err != nil {
-		return nil, errors.EnsureStack(err)
-	}
-	return &config, nil
+	return &connConfig, nil
 }
 
 func (c connectorConfig) toIDPConnector() (*identity.IDPConnector, error) {
-	jsonConfig, err := json.Marshal(c.Config)
+	configBytes, err := json.Marshal(c.Config)
```

Comment:
could you add a comment explaining the JSON/YAML shenanigans?

Reply:
Sure, can do.